src/main/python/deep_qa/layers/entailment_models/tuple_match.py

```diff
@@ -0,0 +1,156 @@
+    where :math:`s` is the index of the slot, :math:`A_s` is answer tuple slot :math:`s` and :math:`B_s` is
+    background tuple slot :math:`s`.
+
+    Inputs:
+        - tuple_1_input (the answer tuple), shape ``(batch size, num_slots, num_slot_words_t1)``,
+          any mask is ignored.  Here num_slot_words_t1 is the maximum number of words in each of the
+          slots in tuple_1.
+        - tuple_2_input (the background_tuple), shape ``(batch size, num_slots, num_slot_words_t2)``,
+          and again, any corresponding mask is ignored. As above, num_slot_words_t2 is the
+          maximum number of words in each of the slots in tuple_2. This need not match tuple 1.
+
+    Output:
+        - entailment score, shape ``(batch, 1)``
+
+    Parameters
+    ----------
+    - num_hidden_layers : int, default=1
+        Number of hidden layers in the shallow NN.
+    - hidden_layer_width : int, default=4
+        The number of nodes in each of the NN hidden layers.
+    - initialization : string, default='glorot_uniform'
+        The initialization of the NN weights
+    - hidden_layer_activation : string, default='relu'
+        The activation of the NN hidden layers
+    - final_activation : string, default='sigmoid'
+        The activation of the NN output layer
+    """
+
+    def __init__(self, num_hidden_layers: int=1, hidden_layer_width: int=4,
+                 initialization: str='glorot_uniform', hidden_layer_activation: str='relu',
+                 final_activation: str='sigmoid'):
+        """
+        Construct a new TupleMatch layer.
+
+        Parameters
+        ----------
+        """
+        self.input_dim = None
+        self.supports_masking = True
+        # Parameters for the shallow neural network
+        self.num_hidden_layers = num_hidden_layers
+        self.hidden_layer_width = hidden_layer_width
+        self.hidden_layer_init = initializations.get(initialization)
+        self.hidden_layer_activation = activations.get(hidden_layer_activation)
+        self.final_activation = activations.get(final_activation)
+        self.hidden_layer_weights = []
+        self.score_layer = None
+        super(TupleMatch, self).__init__()
+
+
+    def get_output_shape_for(self, input_shapes):
+        #pylint: disable=unused-argument
+        return (input_shapes[0], 1)
+
+
+    def build(self, input_shape):
+        super(TupleMatch, self).build(input_shape)
+
+        # Add the weights for the hidden layers
+        hidden_layer_input_dim = input_shape[0][1]
+        for i in range(self.num_hidden_layers):
+            self.hidden_layer_weights.append(self.hidden_layer_init((hidden_layer_input_dim,
+                                                                     self.hidden_layer_width),
+                                                                    name='%s_hiddenlayer_%d' % (self.name, i)))
+            hidden_layer_input_dim = self.hidden_layer_width
+        self.trainable_weights = self.hidden_layer_weights[:]
```

Comment:
You don't need the `[:]`.  Oh, yes you do, otherwise you'd be appending `self.score_layer` to it...  This is a pretty strong argument in favor of using `self.add_weight`, I think.  Less prone to errors like that (not that your code is wrong, just that I would have written it wrong, and I almost suggested to change it to something that's wrong).